### DISCUSSION TITLE
Targeted SHACL Validation

### DISCUSSION THREAD

**Discussion by daltontc** [eclipse-rdf4j/rdf4j#4506] [Ideas]
Is there a way to perform SHACL validation in a more targeted fashion? It appears that right now you load the shapes into a specific named graph then it performs the validation across the whole repository. I think a nice feature to have would be the ability to provide targeted graphs to run SHACL validation against with a set of shapes. Is this something that is feasible?

It seems like the current setup is a way to block transactions that don't adhere to a shape. The ability to generate reports against existing subsets of data would be a nice feature to have.

**Comment by hmottestad** [comment]
You can link a data graph a shapes graph. You can also link to the union of multiple shapes graphs.

https://rdf4j.org/documentation/programming/shacl/#shapes-graph

For validation without using the ShaclSail you can use the Console. Documentation is a bit sparse but I think that `verify data.jsonld shacl-file.ttl validation-report.ttl` should run the SHACL validation and write the report to the specified file. 

In the not too distant future we will support running validation against an existing MemoryStore, NativeStore, etc. without using the ShaclSail.
